src/core/qgsshadingrenderer.h

```diff
@@ -0,0 +1,214 @@
+/***************************************************************************
+  qgsshadingrenderer.h - QgsShadingRenderer
+
+ ---------------------
+ begin                : 4.12.2022
+ copyright            : (C) 2022 by Vincent Cloarec
+ email                : vcloarec at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSSHADINGRENDERER_H
+#define QGSSHADINGRENDERER_H
+
+#include "qdom.h"
+#include "qgsunittypes.h"
+#include "qgis_sip.h"
+#include "qgselevationmap.h"
+
+class QImage;
+class QgsElevationMap;
+class QgsRenderContext;
+class QgsMapSettings;
+class QDomDocument;
+class QgsReadWriteContext;
+
+/**
+ * \ingroup core
+ * \brief This class can render shading on image with different methods (eye dome Lighting, hill shading,...).
+ * Instane od this class support differents settings:
+ * - activate/deactivate the hafind renderer
+ * - activate/deactivate a shading methods
+ * - parameters of each shading methods
+ *
+ * To render shading on image the caller call renderShading() with the image and
+ * an elevation map (see QgsElevationMap()) as parameters. The shading is applied on the image in place.
+ * Elevation map and image must have same sizes.
+ *
+ * \since QGIS 3.30
+ */
+class CORE_EXPORT QgsShadingRenderer
```

Comment:
My suggestion is to rename this to QgsElevationShadingRenderer - to make it clear this is related to elevation data.

Also to make it clearer in the API:
- QgsProject::mapShadingRenderer -> QgsProject::elevationShadingRenderer
- QgsMapSettings::shadingRenderer -> QgsMapSettings::elevationShadingRenderer